public/app/features/scenes/scenes/nestedBasicFromJson.tsx

```diff
@@ -0,0 +1,139 @@
+      range: {
+        from: 'now-6h',
+        to: 'now',
+      },
+    },
+    {
+      key: 'a3ebb3ff-dce8-40bd-8ef3-066fdbdeb7fa',
+      inputParams: {
+        timeRange: '94d745c9-7c8e-4234-bd98-e620ee643a61',
+      },
+      type: 'SceneDataProviderNode',
+      queries: [
+        {
+          refId: 'A',
+          datasource: {
+            uid: 'gdev-testdata',
+            type: 'testdata',
+          },
+          scenarioId: 'random_walk_table',
+        },
+      ],
+    },
+  ],
+  layout: {
+    root: [
+      {
+        direction: 'column',
+        key: '8d42de22-be9a-417e-b42c-634959ab7af4',
+        type: 'SceneFlexLayout',
+        children: [
+          {
+            $ref: 'b30b9a3e-7112-4279-be52-00a913d161a0',
+            type: 'NestedScene',
+          },
+        ],
+      },
+    ],
+    'b30b9a3e-7112-4279-be52-00a913d161a0': {
+      key: 'b30b9a3e-7112-4279-be52-00a913d161a0',
+      title: 'Inner scene',
+      data: 'TODO: serialize Scene data?',
+      inputs: [
+        {
+          key: 'a3ebb3ff-dce8-40bd-8ef3-066fdbdeb7fa',
+          inputParams: {
+            timeRange: '94d745c9-7c8e-4234-bd98-e620ee643a61',
+          },
+          type: 'SceneDataProviderNode',
+          queries: [
+            {
+              refId: 'A',
+              datasource: {
+                uid: 'gdev-testdata',
+                type: 'testdata',
+              },
+              scenarioId: 'random_walk_table',
+            },
+          ],
+        },
+        {
+          key: '94d745c9-7c8e-4234-bd98-e620ee643a61',
+          inputParams: {},
+          type: 'SceneTimeRange',
+          range: {
+            from: 'now-6h',
+            to: 'now',
+          },
+        },
+      ],
+      layout: {
+        root: [
+          {
+            direction: 'row',
+            key: 'b19e3502-3d3d-421d-9659-d32a05598630',
+            type: 'SceneFlexLayout',
+            children: [
+              {
+                key: '45772c3d-57b8-4bbc-b233-7338c720652a',
+                direction: 'row',
+                type: 'SceneFlexChild',
+                children: [
+                  {
+                    inputParams: {
+                      data: 'a3ebb3ff-dce8-40bd-8ef3-066fdbdeb7fa',
+                    },
+                    key: '3',
+                    pluginId: 'timeseries',
+                    title: 'Data',
+                    type: 'VizPanel',
+                  },
+                ],
+              },
+              {
+                key: '1d29a557-8ae9-4891-844d-602d4dccf39f',
+                direction: 'row',
+                type: 'SceneFlexChild',
+                children: [
+                  {
+                    inputParams: {
+                      data: 'a3ebb3ff-dce8-40bd-8ef3-066fdbdeb7fa',
+                    },
+                    key: '3',
+                    pluginId: 'timeseries',
+                    title: 'Data',
+                    type: 'VizPanel',
+                  },
```

Comment:
this panel has an input param for a query data node, but what will it's time range be (if the panel where to call getTimeRange()) ?

Reply:
Ah I see now how getTimeRange is implemented, it will scan all the inputParams and call getTimeRange from them

Reply:
The question is - is the time range relevant for the panel to know about it? Intuitively yest, but in the end the actual object that relies on it is the data node. The time range in the panel/visualization should be derived from the data that it has. But- of course panel can call `this.getTimeRange()` and the "closes" time range will be resolved from the "closes" data node :)